core/tee/tee_fs_rpc.c

```diff
@@ -0,0 +1,600 @@
+
+	/* fill in parameters */
+	*bf = *bf_cmd;
+
+	if (mode & TEE_FS_MODE_IN)
+		memcpy((void *)(bf + 1), data, len);
+
+	ret = thread_rpc_cmd(cmd_id, 1, &params);
+	/* update result */
+	*bf_cmd = *bf;
+	if (ret != TEE_SUCCESS)
+		goto exit;
+
+	if (mode & TEE_FS_MODE_OUT) {
+		uint32_t olen = MIN(len, bf->len);
+
+		memcpy(data, (void *)(bf + 1), olen);
+	}
+
+	res = TEE_SUCCESS;
+
+exit:
+	thread_rpc_free_payload(cpayload);
+	return res;
+}
+
+int tee_fs_rpc_access(int id, const char *name, int mode)
+{
+	struct tee_fs_rpc head = { 0 };
+	TEE_Result res;
+	int rc = RPC_FAILED;
+	size_t len;
+
+	DMSG("(id: %d, name: %s, mode: %d)...", id, name, mode);
+
+	if (!name)
+		goto exit;
+
+	len = strlen(name) + 1;
+	if (len <= 1)
+		goto exit;
+
+	head.op = TEE_FS_ACCESS;
+	head.flags = mode;
+
+	res = tee_fs_rpc_send_cmd(id, &head, (void *)name, len, TEE_FS_MODE_IN);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	rc = head.res;
+exit:
+	DMSG("...%d", rc);
+	return rc;
+}
+
+int tee_fs_rpc_begin_transaction(int id)
+{
+	struct tee_fs_rpc head = { 0 };
+	TEE_Result res;
+	int rc = RPC_FAILED;
+
+	assert(id == OPTEE_MSG_RPC_CMD_SQL_FS);
+
+	DMSG("(id: %d)...", id);
+
+	/* fill in parameters */
+	head.op = TEE_FS_BEGIN;
+	head.fd = -1;
+
+	res = tee_fs_rpc_send_cmd(id, &head, NULL, 0,
+				  TEE_FS_MODE_NONE);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	rc = head.res;
+exit:
+	DMSG("...%d", rc);
+	return rc;
+}
+
+int tee_fs_rpc_close(int id, int fd)
+{
+	struct tee_fs_rpc head = { 0 };
+	TEE_Result res;
+	int rc = RPC_FAILED;
+
+	DMSG("(id: %d, fd: %d)...", id, fd);
+
+	head.op = TEE_FS_CLOSE;
+	head.fd = fd;
+
+	res = tee_fs_rpc_send_cmd(id, &head, NULL, 0, TEE_FS_MODE_NONE);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	rc = head.res;
+exit:
+	DMSG("...%d", rc);
+	return rc;
+}
+
+int tee_fs_rpc_end_transaction(int id, bool rollback)
+{
+	struct tee_fs_rpc head = { 0 };
+	TEE_Result res;
+	int rc = RPC_FAILED;
+
+	assert(id == OPTEE_MSG_RPC_CMD_SQL_FS);
+
+	DMSG("(id: %d, rollback: %d)...", id, rollback);
+
+	head.op = TEE_FS_END;
+	head.arg = rollback;
+	head.fd = -1;
+
+	res = tee_fs_rpc_send_cmd(id, &head, NULL, 0, TEE_FS_MODE_NONE);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	rc = head.res;
+exit:
+	DMSG("...%d", rc);
+	return rc;
+}
+
+int tee_fs_rpc_ftruncate(int id, int fd, tee_fs_off_t length)
+{
+	struct tee_fs_rpc head = { 0 };
+	TEE_Result res;
+	int rc = RPC_FAILED;
+
+	DMSG("(id: %d, fd: %d, length: %" PRId64 ")...", id, fd, length);
+
+	head.op = TEE_FS_TRUNC;
+	head.fd = fd;
+	head.arg = length;
+
+	res = tee_fs_rpc_send_cmd(id, &head, NULL, 0, TEE_FS_MODE_NONE);
+	if (res != TEE_SUCCESS)
+		goto exit;
+
+	rc = head.res;
+exit:
+	DMSG("...%d", rc);
+	return rc;
+}
+
+int tee_fs_rpc_link(int id, const char *old, const char *nw)
+{
+	size_t len_old = strlen(old) + 1;
+	size_t len_new = strlen(nw) + 1;
+	size_t len = len_old + len_new;
+	struct tee_fs_rpc head = { 0 };
+	char *tmp = NULL;
+	TEE_Result res;
+	int rc = RPC_FAILED;
+
```

Comment:
why not checking "old, new" whereas such are tested in `tee_fs_rpc_mkdir`,`tee_fs_rpc_open`